api/src/providers/userDataProvider.ts

```diff
@@ -260,3 +275,50 @@ async function createTriggerRecursively(dbTransaction, alertTriggerId: string, f
     );
   }
 }
+
+export async function enableAlert(userId: string, alertId: string) {
+  const user = await UserSetting.findOne({
+    where: { userId: userId },
+    include: [{ model: UserAlert }]
+  });
+
+  const alert = user.userAlerts.find((x) => x.id === alertId);
+
+  if (!alert) {
+    throw new Error("Alert not found");
+  }
+
+  // Check if user has already enabled alerts
+  const planCode = await getUserPlan(user.stripeCustomerId);
+  const currentPlan = plans.find((x) => x.code === planCode);
+
+  if (user.userAlerts.filter((x) => x.enabled).length + 1 > currentPlan.numberOfAlerts) {
+    throw new Error(`Current user plan only allows ${currentPlan.numberOfAlerts} active alerts.`);
+  }
+
+  await UserAlert.update(
+    {
+      enabled: true
+    },
+    {
+      where: {
+        id: alertId,
+        userId: userId
+      }
+    }
+  );
```

Comment:
Since we already got the alert we can update it directly. It will use the primary key for the `UPDATE` condition so it's probably a tiny bit faster.
```suggestion
  alert.enabled = true;
  await alert.save();
```